src/AmbientSounds.Uwp/Controls/SettingsControl.xaml.cs

```diff
@@ -38,5 +45,45 @@ private void OnThemeSelectionChanged(object sender, SelectionChangedEventArgs e)
                 ViewModel.UpdateTheme(s);
             }
         }
+        
+        // Check if auto-startup is on and adjust its settings accordingly.
+        private async void AutoStartup_Toggled(object sender, Windows.UI.Xaml.RoutedEventArgs e)
+        {
+            StartupTask startupTask = await StartupTask.GetAsync("AmbieAutoStartup");
+            Debug.WriteLine(startupTask.State.ToString());
+            if ((sender as ToggleSwitch).IsOn)
+            {
+                switch (startupTask.State)
+                {
+                    case StartupTaskState.Disabled:
+                        StartupTaskState newState = await startupTask.RequestEnableAsync();
+                        if (newState == StartupTaskState.Enabled || newState == StartupTaskState.EnabledByPolicy)
+                        {
+                            (sender as ToggleSwitch).IsOn = true;
+                        }
+                        else { 
+                            (sender as ToggleSwitch).IsOn = false;
+                        }
+                        break;
+                    case StartupTaskState.DisabledByUser:
+                        // Task is disabled and user must enable it manually.
+                        MessageDialog dialog = new MessageDialog("You need to enable this app's entry in Task Manager.");
+                        await dialog.ShowAsync();
+                        (sender as ToggleSwitch).IsOn = false;
+                        break;
+                    case StartupTaskState.DisabledByPolicy:
+                        MessageDialog dialog2 = new MessageDialog("Startup disabled by group policy, or not supported on this device");
```

Comment:
Use ContentDialog instead maybe? MessageDialog has the old metro design and isn't regularly used in UWP.

Reply:
true, I confuse the two often. thanks for the heads-up.